### PR TITLE
Simulator Execution deposits and withdrawals

### PR DESCRIPTION
**Pull request by javcode**
- Added Simulator Execute Deposit API [Docs](https://335-merge--jovial-scone-ec740d.netlify.app/api-reference/execute-simulator-deposit/)
- Added Simulator Execute Withdrawal API [Docs](https://335-merge--jovial-scone-ec740d.netlify.app/api-reference/execute-simulator-withdrawal/)
- Added `simulator_call` instruction to spending pre-requisites [endpoint](https://335-merge--jovial-scone-ec740d.netlify.app/api-reference/get-spending-prerequisites/)
- Added withdrawal execution instruction for funding source using simulator strategy. Endpoint [here](https://335-merge--jovial-scone-ec740d.netlify.app/api-reference/create-withdrawal-intent/)

Preview [here](https://335-merge--jovial-scone-ec740d.netlify.app)

Ticket Link: https://www.notion.so/immersve/public-documentation-ec1b7d9efe8d4392918b7f1d28054d51?pvs=4

### CODE REVIEW

File: imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+post:
+  tags:
+    - simulator
+  summary: Execute simulator deposit
+  description: |
+    This endpoint can be used to simulate a deposit and use it as a source of funds.
```
Comment (javcode) [inline imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml:6]:
```suggestion
    Simulate a deposit and use it as a source of funds.
```

File: imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      description: Successful operation
+      content:
+        application/json:
+          schema:
+            $ref: "../funding-sources/models/funding-source-interaction.yaml"
+    "403":
+      description: |
+        User is not allowed to perform the action with the reason stated in the `errorCode`
+
+        **FORBIDDEN**
```
Comment (javcode) [inline imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml:25]: review, it shouldn't be necessary

File: imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        Insufficient permissions to perform the action.
+
+        **INVALID_FUNDING_SOURCE_TYPE**
+        Funding source type is invalid. Only funding types using the Simulator strategies are valid
+      content:
+        application/json:
+          schema:
+            $ref: "../../models/api-error-403.yaml"
+
+    "400":
```
Comment (javcode) [inline imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml:35]: see if we put this in a common place

File: imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-withdrawal.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+post:
+  tags:
+    - simulator
+  summary: Execute simulator withdrawal
+  description: |
+    This endpoint can be used to simulate a withdrawal on a simulated funding source
```
Comment (javcode) [inline imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-withdrawal.yaml:6]:
```suggestion
    Simulate a withdrawal on a simulated funding source
```

File: imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-withdrawal.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            $ref: "../funding-sources/models/funding-source-interaction.yaml"
+    "403":
+      description: |
+        User is not allowed to perform the action with the reason stated in the `errorCode`
+
+        **FORBIDDEN**
+        Insufficient permissions to perform the action.
+
+        **INVALID_FUNDING_SOURCE_TYPE**
+        Funding source type is invalid. Only funding types using the Simulator strategies are valid
```
Comment (javcode) [inline imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-withdrawal.yaml:29]:
```suggestion
        Funding source type is invalid. Only funding types using the simulator strategies are valid.
```

File: imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            $ref: "../funding-sources/models/funding-source-interaction.yaml"
+    "403":
+      description: |
+        User is not allowed to perform the action with the reason stated in the `errorCode`
+
+        **FORBIDDEN**
+        Insufficient permissions to perform the action.
+
+        **INVALID_FUNDING_SOURCE_TYPE**
+        Funding source type is invalid. Only funding types using the Simulator strategies are valid
```
Comment (javcode) [inline imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml:29]:
```suggestion
        Funding source type is invalid. Only funding types using the simulator strategies are valid.
```

File: imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        schema:
+          $ref: "./models/execute-deposit-request.yaml"
+    required: true
+  responses:
+    "200":
+      description: Successful operation
+      content:
+        application/json:
+          schema:
+            $ref: "../funding-sources/models/funding-source-interaction.yaml"
```
Comment (javcode) [inline imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml:20]: make it specific to this endpoint. Is Withdrawal type, without context

File: imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml (hunk 2 above)
Comment (javcode) [reply imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml:25]: other endpooints are doing the same, leave it for now

File: imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml (hunk 3 above)
Comment (javcode) [reply imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml:35]: again, other endpoints are doing it the same way, leave it for now

File: imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml (hunk 7 above)
Comment (javcode) [reply imsv-docs-docusaurus/openapi/endpoints/simulator/execute-simulator-deposit.yaml:20]: updated

**Review by ncjones — APPROVED** [review]

**Pull request merged**